tornado/routing.py

```diff
@@ -286,10 +297,10 @@ def finish(self) -> None:
 
 # _RuleList can either contain pre-constructed Rules or a sequence of
 # arguments to be passed to the Rule constructor.
-_RuleList = List[
+_RuleList = Sequence[
     Union[
         "Rule",
-        List[Any],  # Can't do detailed typechecking of lists.
+        Sequence[Any],  # Can't do detailed typechecking of lists.
```

Comment:
What happens if we only change the outer List to Sequence? I don't think we need covariance on the inner `List[Any]`, and then we wouldn't need to change the `(tuple, list)` typecheck below. (Logically, this inner sequence is only supposed to be 2-, 3-, or 4-tuples. For historical reasons we also allow lists of the same lengths, but I don't see any reason to extend this to arbitrary sequences).